src/view/context/view.cpp

```diff
@@ -1,19 +1,55 @@
 #include "view/context/view.hpp"
 #include "mainwindow.hpp"
 
-Context::View::View(lib::spt::api &spotify, spt::Current &current,
+Context::View::View(lib::spt::api &spotify, lib::settings &settings, spt::Current &current,
 	const lib::cache &cache, QWidget *parent)
-	: QDockWidget(parent)
+	: QDockWidget(parent),
+	spotify(spotify),
+	current(current),
+	cache(cache)
 {
-	content = new Context::Content(spotify, current, cache, this);
-	setWidget(content);
+	// check if user wants a scalable album image or not
+	albumShouldBeExpandable = settings.general.expand_album_cover;
+	if (albumShouldBeExpandable)
+	{
+		horizContent = new Context::HorizContent(spotify, current, cache, this);
+		setWidget(horizContent);
+	}
+	else 
+	{
+		content = new Context::Content(spotify, current, cache, this);
+		setWidget(content);
+	}
 
 	title = new Context::Title(spotify, current, cache, this);
 	setTitleBarWidget(title);
 
 	setFeatures(QDockWidget::DockWidgetMovable | DockWidgetFloatable);
 }
 
+void Context::View::reloadAlbumContent(bool shouldBeExpandable)
+{
+	albumShouldBeExpandable = shouldBeExpandable;
+	if (content == nullptr || horizContent == nullptr)
+	{
+		free(content);
+		free(horizContent);
+		content = nullptr;
+		horizContent = nullptr;
```

Comment:
Like I feel like this is not the optimal way to perform object updates. It just feels wrong - but IDK how else to do something like this in C++ :pleading_face:

Reply:
`free`, and `malloc` to allocate memory, is the "C way" of doing memory management, while C++ prefers using `delete` and `new` respectively. However, in Qt, you very rarely need to do manual memory management. If you manually need to delete something, like when switching out a widget or similar, you can `QObject::deleteLater` instead.